packages/shared/scope-hoisting/src/link.js

```diff
@@ -663,6 +689,13 @@ export function link({
           options,
         );
 
+        if (
+          process.env.PARCEL_BUILD_ENV !== 'production' &&
+          bundle.env.outputFormat !== 'esmodule'
```

Comment:
why not ESM?

Reply:
See here:
https://github.com/parcel-bundler/parcel/pull/4271/files/1a6cb5d837c05061045e96e12b4950a2934e11eb#diff-5ea34835db03f5baab5cf045aed280c1

But I've made updated the check to just recrawl during development after the problematic modification (and verify beforehand).